Samples/NetworkMediaStreamer/RTPMediaStreamer/src/RTPStreamSink.cpp

```diff
@@ -0,0 +1,443 @@
+        }
+
+        sockaddr_in Server;
+        Server.sin_family = AF_INET;
+        Server.sin_addr.s_addr = INADDR_ANY;
+        for (u_short P = m_localRTPPort; P < 0xFFFE; P += 2)
+        {
+            m_RtpSocket = socket(AF_INET, SOCK_DGRAM, 0);
+            Server.sin_port = htons(P);
+            if (bind(m_RtpSocket, (sockaddr*)&Server, sizeof(Server)) == 0)
+            {   // Rtp socket was bound successfully. Lets try to bind the consecutive Rtsp socket
+                m_RtcpSocket = socket(AF_INET, SOCK_DGRAM, 0);
+                Server.sin_port = htons(P + 1);
+                if (bind(m_RtcpSocket, (sockaddr*)&Server, sizeof(Server)) == 0)
+                {
+                    m_localRTPPort = P;
+                    m_localRTCPPort = P + 1;
+                    break;
+                }
+                else
+                {
+                    closesocket(m_RtpSocket);
+                    m_RtpSocket = INVALID_SOCKET;
+                    closesocket(m_RtcpSocket);
+                    m_RtcpSocket = INVALID_SOCKET;
+                }
+            }
+            else
+            {
+                closesocket(m_RtpSocket);
+                m_RtpSocket = INVALID_SOCKET;
+            }
+        }
+
+        inet_pton(AF_INET, destination.c_str(), &(m_remoteAddr.sin_addr));
+        m_remoteAddr.sin_port = htons(m_remotePort);
+        m_remoteAddr.sin_family = AF_INET;
+    }
+}
+
+TxContext::~TxContext()
+{
+    if (m_RtpSocket != INVALID_SOCKET)
+    {
+        closesocket(m_RtpSocket);
+    }
+    if (m_RtcpSocket != INVALID_SOCKET)
+    {
+        closesocket(m_RtcpSocket);
+    }
+}
+
+void TxContext::SendPacket(winrt::Windows::Storage::Streams::IBuffer buf)
+{
+    if (m_packetHandler)
+    {
+        m_packetHandler(nullptr, buf);
+    }
+    else
+    {
+        auto pBuf = buf.data();
+        auto sz = buf.Length();
+        sendto(m_RtpSocket, (char*)pBuf, (int)sz, 0, (SOCKADDR*)&m_remoteAddr, sizeof(m_remoteAddr));
+    }
+    m_SequenceNumber++;
+}
+
+
+RTPVideoStreamSink::RTPVideoStreamSink(IMFMediaType* pMediaType, IMFMediaSink* pParent, DWORD dwStreamID)
+    : NwMediaStreamSinkBase(pMediaType, pParent, dwStreamID)
+    , m_packetizationMode(1)
+    , m_SequenceNumber(0)
+    , m_pTxBuf(nullptr)
+{
+    if (m_packetizationMode == 1)
+    {
+        m_MTUSize = 1500;
+    }
+    else
+    {
+        //TODO: find a way to control encoder's max NAL size and edit this param to a better value
+        m_MTUSize = 65535; // max size to allow any size NAL into one packet
+    }
+    m_pTxBuf = winrt::Windows::Storage::Streams::Buffer((uint32_t)m_MTUSize);
+
+}
+
+BYTE* RTPVideoStreamSink::FindSC(BYTE* bufStart, BYTE* bufEnd)
+{
+    auto it = bufStart;
+    while (it < bufEnd - 3)
+    {
+        if ((it[0] == 0) && (it[1] == 0) && (it[2] == 1)
+            || (*((uint32_t*)it) == 0x01000000)
+            )
+        {
+            return it;
+        }
+        while (*(++it) != 0);

```

Comment:
>while (*(++it) != 0); [](start = 8, length = 21)

suggestion: I would avoid a while with a condition that precludes the other while's condition like that (which in theory can read beyond bufEnd). In your while loop (or a for) I would use a first 'if' condition and then in that first if context you use a second if for the other conditions, which if both are true then you return. Else you just increment it (it++)
...
    for (auto it = bufStart; it < bufEnd - 3; it++)
    {
        if ((it[0] == 0) 
        {
            if ((it[1] == 0) && (it[2] == 1)
              || (*((uint32_t*)it) == 0x01000000))
            {
                return it;
            }
        }
    return bufEnd;
}